### PR TITLE
feat(auth): implement registration, refactor validation and add unit tests

### PR DESCRIPTION
**Pull request by FabianDobber**
**Description:**
This pull request implements the full user registration flow (US24) and updates the login mechanism (US23). It refactors input validation logic from the ViewModel into a dedicated Core helper to ensure testability and separation of concerns. Additionally, comprehensive unit tests have been added for authentication services and validation logic using NUnit and Moq.

**Registration and Login Logic**
*   A new `RegisterView` and `RegisterViewModel` have been implemented, allowing users to create accounts with specific roles (Lid, Nieuw Lid, Gast).
*   The `LoginViewModel` has been updated to handle role-based routing. Guests are currently restricted with a notification, while Members are routed to the boat selection screen.
*   The `AuthService` has been expanded to support registration, check for existing emails, and retrieve user roles.
*   The `RoleRepository` now automatically seeds default roles if the database is empty.

**Refactoring & Validation**
*   Input validation logic (Regex for names/emails, range checks for rowing levels) has been extracted from the `RegisterViewModel` into a new `ValidationHelper` class in the Core layer.
*   This change decouples the UI from business rules, allowing for direct unit testing of validation requirements (FR2-T, NFR1) without relying on the UI layer.

**Unit Test Coverage**
*   **ValidationHelperTests:** Comprehensive tests covering all edge cases for input validation (null values, whitespace handling, regex patterns, and boundary values).
*   **AuthServiceTests:** Tests using Moq to verify login credentials, registration data integrity, and role retrieval logic.
*   **Result:** 100% pass rate on all new tests.

image

**Links**
*   [US23 - Inloggen Gebruiker](https://trello.com/c/4zp5S9Hw/71-us23-userstory-inloggen-gebruiker)
*   [US24 - Registreren Gebruiker](https://trello.com/c/MNb1TLse/93-us24-userstory-registreren-gebruiker)

### CODE REVIEW

File: ProjectBotenReservering.App/AppShell.xaml
Comment (Skypoop) [inline ProjectBotenReservering.App/AppShell.xaml]: App now crashes when clicking login due to changes regarding the sidebar. Has been discussed, problem is not limited to this file. Please fix.

File: ProjectBotenReservering.App/App.xaml.cs [hunk 1]
```diff
@@ -7,8 +7,10 @@ public partial class App : Application
         public App(LoginView loginView)
         {
             InitializeComponent();
-            MainPage = loginView;
-
+        }
+        protected override Window CreateWindow(IActivationState? activationState)
+        {
+            return new Window(new AppShell());
```
Comment (Skypoop) [inline ProjectBotenReservering.App/App.xaml.cs]: This refactor may be contributing to the mentioned crash when clicking "login". Please explain why this protected override is necessary, or revert it.

File: ProjectBotenReservering.Core/Services/AuthService.cs [hunk 2]
```diff
@@ -1,22 +1,50 @@
...
+        _clientRepository.Add(newClient);
+        _clientRoleRepository.Add(new ClientRole(roleName, newClient.Id));
+
+        return true;
+    }
+
+    public string GetUserRole(int clientId)
+    {
+        List<ClientRole> roles = _clientRoleRepository.GetByClientId(clientId);
+        ClientRole clientRole = roles.FirstOrDefault();
```
Comment (Skypoop) [inline ProjectBotenReservering.Core/Services/AuthService.cs]: Converting null literal or possible null value to non-nullable type.
Replace "ClientRole" with "ClientRole?"
